spyder/plugins/completion/kite/providers/document.py

```diff
@@ -46,11 +47,14 @@ def convert_text_snippet(snippet_info):
         placeholder_end = placeholder['end']
         next_pos = placeholder_begin
         standard_text = text[prev_pos:next_pos]
-        snippet_text = text[next_pos:placeholder_end][1:-1]
+        snippet_text = text[next_pos:placeholder_end]
         prev_pos = placeholder['end']
         text_builder.append(standard_text)
         placeholder_number = (i + 1) % total_placeholders
-        snippet = '${%d:%s}' % (placeholder_number, snippet_text)
+        if snippet_text:
+            snippet = '${%d:%s}' % (placeholder_number, snippet_text)
+        else:
+            snippet = '$%d' % (placeholder_number)
```

Comment:
I think this is redundant by the change introduced on line 56

Reply:
We were not previously calling `convert_text_snippet` on non-children completions below (see my change below on new line 129). Now that we are, it bubbled up a bug, where placeholder text returned by Kite may be empty.

If we use `${1:}` for empty placeholder text, the snippet parser doesn't fail outright, but an exception is thrown when trying to insert the snippet. The correct syntax for placeholders without text is `$1`.

As far as I can tell, putting `$0` at the end, as you do on line 56 only fixes the final tabstop. Let me know if I'm missing something here.

Reply:
So, this should cover cases when the cursor lands inside a function call brackets?

Reply:
that's right (there are also some other cases where Kite currently returns empty placeholders / tabstops without text).

Reply:
Then, we're good with this one!